src/main/java/seedu/financeit/goaltracker/GoalTracker.java

```diff
@@ -165,36 +267,120 @@ public static void targetGoalTracker(Entry entry) {
         }
     }
 
+    public static void storeLedgerDate(Ledger ledger) {
+        ledgerMonth = ledger.getDate().getMonth();
+    }
+
     public static void handleExpenseGoal() {
         LedgerList ledgerList = ManualTracker.getLedgerList();
+        totalExpenses = 0;
         int ledgerSize = ledgerList.getItemsSize();
-        for (int i = ledgerSize - 1; i < ledgerSize; i++) {
+        for (int i = 0; i < ledgerSize; i++) {
             Ledger ledger = (Ledger) ledgerList.getItemAtIndex(i);
-            EntryList entryList = ledger.entryList;
-            int entrySize = entryList.getItemsSize();
-            for (int x = entrySize - 1; x < entrySize; x++) {
-                expenses = entryData.getAmount();
-                totalExpenses += expenses;
+            if (ledgerMonth == null) {
+                month = Month.of(Integer.parseInt(cmdPacket[3]));
+                if (ledger.getDate().getMonth().equals(month)) {
+                    EntryList entryList = ledger.entryList;
+                    int entrySize = entryList.getItemsSize();
+                    for (int x = 0; x < entrySize; x++) {
+                        Entry entry = (Entry) entryList.getItemAtIndex(x);
+                        if (entry.getEntryType().toString().equals("Expense")) {
+                            expenses = entry.getAmount();
+                            totalExpenses += expenses;
+                        } else {
+                            totalExpenses += totalExpenses;
+                        }
+                    }
+                }
```

Comment:
Consider storing total monthly expenditure/income in a hashMap.
E.g.
expenditureList
JAN 1500
FEB 254
MAR 0

Increment/decrement the total amount whenever there is a change made. For instance, when a new expense of $25 is added to the ledger for June 15 2020, increment the total expenditure for June by $25. When this same expense is edited to become $45, increment June expenditure by $20. When this expense is deleted, decrement June expenditure by $45.

This way, there is no need to loop through all entries every time there is a new entry or every time you want to view the goal.